src/C-DEngine/C-DCommunication/Protocols/WS/cde_WSCommon.cs

```diff
@@ -147,7 +147,21 @@ internal void ProcessIncomingData(string pPostString, byte[] pPostData, int pPos
                 return;
             }
             TheRequestData tRequestData = TheRequestData.CloneForWS(MySessionRequestData);
+
```

Comment:
Please add the EnableKPI if block here as well. I don't think its a boolean in the TheWSProcessorBase, yet. Can you make it similar to TheQueuedSender best in the "SetRequest" function as that has to come before any other call and is not high performance like the "ProcessIncomingData" function

Reply:
This is good point.
I tried to solve that by adding an internal property `EnableKpi` in `TheCDEKpis` which is fully handled there. So any call to any of the `SetKPI`, `GetKPI` oder `De-/IncrementKpi` whill consider the setting. Please check whether this is enough for the high performance use cases. At least it catches all "KPI calls".